### PR TITLE
enhance: Enable prefer node lables in resource group

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: milvus-io/milvus#36977
add prefer node labels in resource groups, so resource group will prefer to accept some node with given label.

### CODE REVIEW

File: proto/rg.proto [hunk 1]
```diff
@@ -19,10 +19,15 @@ message ResourceGroupTransfer {
   string resource_group = 1; // resource groups can be transfered with current resource group.
   // preserve for other option, such as weight, priority or affinity setup.
 }
+
+message ResourceGroupNodeFilter {
+  repeated string prefer_node_labels = 1; // node in resource group must match node labels requirements
```
Comment (chyezh) [inline proto/rg.proto:24]: Does our label system support key-value in future?
QueryNodes may be heterogeneous and have hardware label; they may have user label in multi-tenant system; values ​​for different type identifiers may conflict.
I recommend referring to k8s's selector/label system to preserve future scalability.

https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

**Review by congqixia — COMMENTED** [review]
node label selector has only key value pairs, which means only `equal` operator is supported. Do we need operator to support "not equal", "in", "not in", etc?

**Comment by weiliu1031** [comment]
> node label selector has only key value pairs, which means only `equal` operator is supported. Do we need operator to support "not equal", "in", "not in", etc?

resource group only need label set selector for now

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**